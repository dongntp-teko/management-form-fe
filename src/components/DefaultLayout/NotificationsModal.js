// @flow
import React from 'react';
import { Modal, Table } from 'antd';
import { dateTimeHelper } from 'helpers';
import { defaultConfigs, notificationConstants } from 'constant';
import type { NotificationT } from 'types/notification';

type PropT = {
  visible: boolean,
  onCancel: Function,
  notifications: Array<NotificationT>,
}


const columns = [
  {
    title: 'Tiêu đề',
    dataIndex: 'title',
  },
  {
    title: 'Nội dung',
    dataIndex: 'message',
  },
  {
    title: 'Type',
    dataIndex: 'type',
    filters: Object.values(notificationConstants.types),
    onFilter: (value, record) => record.type.indexOf(value) === 0,
    render: type => <span className={`p-2 badge badge-${type}`}>{type}</span>,
  },
  {
    title: 'Thời gian tạo',
    dataIndex: 'createdAt',
    render: createdAt => dateTimeHelper.formatDateTime(createdAt),
  },
];

export default ({visible, onCancel, notifications}: PropT) => {
  const tableConfigs = {
    ...defaultConfigs.antTable, locale: {
      emptyText: 'Không có thông báo',
    },
  };
  return (
    <Modal
      title={<div className="text-center">Thông báo</div>}
      centered
      closable
      footer={null}
      onCancel={onCancel}
      visible={visible}
      width={1000}
    >
      <Table
        dataSource={notifications}
        columns={columns}
        {...tableConfigs}
        rowKey="id"
      />;
    </Modal>
  );
};
