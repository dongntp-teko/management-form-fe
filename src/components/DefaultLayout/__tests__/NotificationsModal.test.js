// Created by thanhpd on 5/24/2019
// @flow

import React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import NotificationsModal from '../NotificationsModal';


const props = {
  visible: true,
  onCancel: () => {},
  notifications: [
    {
      id: 1,
      title: 'hello',
      message: 'hello',
      type: 'success',
      createdAt: new Date().toISOString(),
      seen: false,
    },
  ],
};

describe('test NotificationsModal component', () => {
  it('should render correctly', () => {
    const {container} = render(<NotificationsModal {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
