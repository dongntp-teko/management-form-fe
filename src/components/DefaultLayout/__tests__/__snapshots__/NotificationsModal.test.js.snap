// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NotificationsModal component should render correctly 1`] = `null`;
